### ISSUE TITLE
TypeError: keras_model_memory_usage() missing 1 required positional argument: 'model'

### ISSUE THREAD

**Issue by farcaz** [Mr-TalhaIlyas/Tensorflow-Keras-Model-Profiler#1]
I have a resnet based model which I'm trying to examine with your script but this error is cracking me up. I tried to even look into the source but everything seems fine. Can you please help?
Copying the traceback here:
.
.
Resnet_profile = model_profiler( resnet_model, 64)
  File "/home/farhat/.virtualenvs/dl4cv/lib/python3.6/site-packages/model_profiler/profiler.py", line 40, in model_profiler
    mem = keras_model_memory_usage(use_units[2], model, Batch_size)
  File "/home/farhat/.virtualenvs/dl4cv/lib/python3.6/site-packages/model_profiler/utils.py", line 22, in deco
    return multiplier.get(units, -1) * fn(*args, **kwds)
  File "/home/farhat/.virtualenvs/dl4cv/lib/python3.6/site-packages/model_profiler/utils.py", line 72, in keras_model_memory_usage
    layer, batch_size=batch_size
  File "/home/farhat/.virtualenvs/dl4cv/lib/python3.6/site-packages/model_profiler/utils.py", line 22, in deco
    return multiplier.get(units, -1) * fn(*args, **kwds)
TypeError: keras_model_memory_usage() missing 1 required positional argument: 'model'

**Comment by Mr-TalhaIlyas** [comment]
can you please share the code so that I can reproduce the error. I tried to regenerate the error form your log but got a completely different error. Some layers were not added in the calculation. But now I resolved that error.

So, now if you can share the code then I'll fix that one also and upload the latest version together.

**Comment by farcaz** [comment]
Here's the code which I've used to test the model profiler:
`import tensorflow as tf
from tensorflow.keras.applications import ResNet50
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout, Flatten
from tensorflow.keras.datasets import mnist
import numpy as np
from tensorflow.python.client import device_lib
from model_profiler import model_profiler

#Resnet feature representation for MNIST
(x_train, y_train), (x_test, y_test) = tf.keras.datasets.mnist.load_data()

# expand new axis, channel axis 
x_train = np.expand_dims(x_train, axis=-1)
x_test = np.expand_dims(x_test, axis=-1)

# [optional]: we may need 3 channel (instead of 1)
x_train = np.repeat(x_train, 3, axis=-1)
x_test = np.repeat(x_test, 3, axis=-1)

# it's always better to normalize 
x_train = x_train.astype('float32') / 255
x_test = x_test.astype('float32') / 255

# resize the input shape , i.e. old shape: 28, new shape: 32
x_train = tf.image.resize(x_train, [32,32]) # if we want to resize
x_test = tf.image.resize(x_test, [32,32]) # if we want to resize 

# one hot 
y_train_hot = tf.keras.utils.to_categorical(y_train , num_classes=10)
y_test_hot = tf.keras.utils.to_categorical(y_test , num_classes=10)

    
resnet_model = Sequential()
resnet_model.add(ResNet50(weights='imagenet',include_top=False, pooling='avg'))

resnet_model.add(Dropout(0.50))
resnet_model.add(Dense(10, activation='softmax'))

# model.layers[0].trainable = False

resnet_model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])
ResNet_H=resnet_model.fit(x_train,y_train_hot,validation_data=(x_test,y_test_hot),batch_size=64,epochs=5,verbose=1)
#x_test= x_test.reshape(-1,32,32,3)
features = resnet_model.predict(x_test)
print(features.shape[0])
#features = features.reshape((features.shape[0], 1 * 1 * 2048))
print(features.shape)
#Resnet Model profiling
print(resnet_model.count_params())
Resnet_profile = model_profiler( resnet_model, 64)`

**Comment by nimaa77** [comment]
problem is here 
https://github.com/Mr-TalhaIlyas/Tensorflow-Keras-Model-Profiler/blob/f90609cb61b3afbe702fc25b7a19dcf6e72dd92f/model_profiler/utils.py#L50
this function takes 2 arguments

but here we are calling it we 3 args 

https://github.com/Mr-TalhaIlyas/Tensorflow-Keras-Model-Profiler/blob/master/model_profiler/profiler.py#L40

**Comment by farcaz** [comment]
Even if I change the function call in Line 40 in profile.py to just 2 arguments(model, batch_size), I get the following error: 
Traceback (most recent call last):
  File "modelProfileTest.py", line 54, in <module>
    Resnet_profile = model_profiler( resnet_model, 64)
  File "/home/farhat/.virtualenvs/dl4cv/lib/python3.6/site-packages/model_profiler/profiler.py", line 40, in model_profiler
    mem = keras_model_memory_usage(model, Batch_size)
  File "/home/farhat/.virtualenvs/dl4cv/lib/python3.6/site-packages/model_profiler/utils.py", line 22, in deco
    return multiplier.get(units, -1) * fn(*args, **kwds)
TypeError: keras_model_memory_usage() missing 1 required positional argument: 'batch_size'

**Comment by farcaz** [comment]
One possible remedy to this situation is to detach the 'keras_model_memory_usage(model, batch_size)' function from the multiplier which actually solved my problem, but I'm not sure if you want that.
To summarize the changes:
1. I removed the @multiply from above the `def keras_model_memory_usage(model, batch_size):`
2. I uncommented the actual memory calculation code at the bottom of the function before return.
3. I changed the function call from profiler.py to simply take 2 arguments i.e model and batch_size
`mem=keras_model_memory_usage(model, batch_size)`
The output I received is like this:
![image](https://user-images.githubusercontent.com/12154952/125162437-62a3ca00-e1ba-11eb-92a9-7ccf6c1930cf.png)

**Comment by farcaz** [comment]
@Mr-TalhaIlyas With your permission, I'm closing this issue.
Thank you for this repository. I would recommend others also to use this for an easy, one place solution related to model computational analysis.
Thanks!

**Comment by Mr-TalhaIlyas** [comment]
Hi there, sorry for extremely late response. I was a bit busy. Anyways I have tried to solve the above mentioned issues. @farcaz  actually your model was built upon `keras` application so the first layer was a `keras functional engine` object and the `lib` was treating it as a layer which caused the error. The error was quite misleading. As for the function call mentioned by @nimaa77 , the function has a `decorator` so the it was called correctly. The problem was it was not designed to handle the combination of custom layers added on top of keras builtin model. Now your above model will work fine I changed the method to handle the models which are built on top of `keras.application`. Also there was some variable overflowing error that is also fixed in version `1.1.8` you can try that to see if the problem is solved.

```python
from tensorflow.keras.applications import ResNet50
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout

from model_profiler import model_profiler

custom_model = Sequential()
custom_model.add(ResNet50(weights='imagenet', include_top=False, input_shape=(32,32,3), pooling='avg'))

custom_model.add(Dropout(0.50))
custom_model.add(Dense(10, activation='softmax'))
units = ['GPU IDs', 'MFLOPs', 'GB', 'Million', 'MB']

profile = model_profiler(model=custom_model, Batch_size=128, use_units=units)
```

output 

```
| Model Profile                    | Value         | Unit    |
|----------------------------------|---------------|---------|
| Selected GPUs                    | None Detected | GPU IDs |
| No. of FLOPs                     | 1.5745        | MFLOPs  |
| GPU Memory Requirement           | 0.3875        | GB      |
| Model Parameters                 | 23.6082       | Million |
| Memory Required by Model Weights | 90.0582       | MB      |
```
